pkg/template/servicebroker/bind.go

```diff
@@ -0,0 +1,131 @@
+			continue
+		}
+
+		name := makeEnvVariableName(service.Name) + "_SERVICE_HOST"
+		services[name] = service.Spec.ClusterIP
+
+		name = makeEnvVariableName(service.Name) + "_SERVICE_PORT"
+		services[name] = strconv.Itoa(int(service.Spec.Ports[0].Port))
+
+		for _, port := range service.Spec.Ports {
+			if port.Name != "" {
+				services[name+"_"+makeEnvVariableName(port.Name)] = strconv.Itoa(int(port.Port))
+			}
+		}
+	}
+
+	return services, nil
+}
+
+func (b *Broker) getSecrets(namespace, instance_id string) (map[string]interface{}, *openservicebroker.Response) {
+	requirement, _ := labels.NewRequirement(templateapi.TemplateInstanceLabel, selection.Equals, []string{instance_id})
+
+	secretList, err := b.kc.Secrets(namespace).List(kapi.ListOptions{LabelSelector: labels.NewSelector().Add(*requirement)})
+	if err != nil {
+		return nil, openservicebroker.NewResponse(http.StatusInternalServerError, nil, err)
+	}
+
+	secrets := map[string]interface{}{}
+	for _, secret := range secretList.Items {
+		if secret.Type != kapi.SecretTypeBasicAuth {
+			continue
+		}
+
+		name := makeEnvVariableName(secret.Name + "_" + kapi.BasicAuthUsernameKey)
+		secrets[name] = string(secret.Data[kapi.BasicAuthUsernameKey])
+
+		name = makeEnvVariableName(secret.Name + "_" + kapi.BasicAuthPasswordKey)
+		secrets[name] = string(secret.Data[kapi.BasicAuthPasswordKey])
+	}
+
+	return secrets, nil
+}
+
+func (b *Broker) Bind(instance_id, binding_id string, breq *openservicebroker.BindRequest) *openservicebroker.Response {
+	if len(breq.Parameters) != 0 {
+		return openservicebroker.NewResponse(http.StatusBadRequest, nil, errors.New("parameters not supported on bind"))
+	}
+
+	uuidmap, err := b.oc.TemplateBrokerUUIDMaps().Get(instance_id)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return openservicebroker.NewResponse(http.StatusBadRequest, nil, err)
+		}
+
+		return openservicebroker.NewResponse(http.StatusInternalServerError, nil, err)
+	}
+
+	ti, err := b.oc.TemplateInstances(uuidmap.Spec.TemplateInstanceRef.Namespace).Get(uuidmap.Spec.TemplateInstanceRef.Name)
+	if err != nil {
+		return openservicebroker.NewResponse(http.StatusInternalServerError, nil, err)
+	}
+	if breq.ServiceID != string(ti.Spec.Template.UID) {
+		return openservicebroker.NewResponse(http.StatusBadRequest, nil, errors.New("service_id does not match provisioned service"))
+	}
+
+	status := http.StatusCreated
+	for _, id := range uuidmap.Spec.BindingIDs {
+		if id == binding_id {
+			status = http.StatusOK
+			break
+		}
+	}
+	if status == http.StatusCreated {
+		uuidmap.Spec.BindingIDs = append(uuidmap.Spec.BindingIDs, binding_id)
+		uuidmap, err = b.oc.TemplateBrokerUUIDMaps().Update(uuidmap)
+		if err != nil {
+			return openservicebroker.NewResponse(http.StatusInternalServerError, nil, err)
+		}
+	}
+
+	namespace := uuidmap.Spec.TemplateInstanceRef.Namespace
+
+	services, resp := b.getServices(namespace, instance_id)
```

Comment:
OK - as a starting point I will have the bind call read the impersonation user from its parameters.